pkg/lib/cockpit-fsinfo.ts

```diff
@@ -0,0 +1,137 @@
+"use strict";
+
+function is_json_dict(value: cockpit.JsonValue): value is cockpit.JsonObject {
```

Comment:
What does this type annotation mean? It feels both wrong (the function returns a bool) and inconsistent (assigning two different types to `value`). It may be right, and it may just be how TypeScript works, but for a newbie like me it looks really irritating.

Reply:
The name of the function is a good hint there.  This is a user-defined type guard function.  In general functions that look like

```ts
function is_x(value): value is X {
  ...
}
```

mean that this function takes a value and dynamically checks if it satisfies the given static type.  The `is` in the return type annotation means that this is a boolean that's true when the predicate is satisfied.

TS uses that so you can say things like

```ts
if (is_x (value)) {
  ...
}
```

and it then knows the type of `value` inside of the block.

Reply:
See https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
Ah, that helped, thank you! :hugs: